flask_icu/__init__.py

```diff
@@ -0,0 +1,454 @@
+        behaves as if a function was registered that returns `None` all the
+        time.  If `None` is returned, the timezone falls back to the one from
+        the configuration.
+
+        This has to return the timezone as string (eg: ``'Europe/Vienna'``)
+        """
+        assert self.timezone_selector_func is None, \
+            'a timezoneselector function is already registered'
+        self.timezone_selector_func = f
+        return f
+
+
+    def list_translations(self):
+        """Returns a list of all the locales translations exist for.  The
+        list returned will be filled with actual locale objects and not just
+        strings.
+        """
+        dirname = os.path.join(self.app.root_path, TRANSLATIONS_PATH)
+        if not os.path.isdir(dirname):
+            return []
+        return [name for name in os.listdir(dirname)
+            if os.path.isdir(os.path.join(dirname, name))]
+
+    @property
+    def default_locale(self):
+        """The default locale from the configuration as instance of a
+        `icu.Locale` object.
+        """
+        default = self.app.config['ICU_DEFAULT_LOCALE']
+        if default is None:
+            default = 'en'
+        return Locale(default)
+
+    @property
+    def default_timezone(self):
+        """The default timezone from the configuration as instance of a
+        `icu.TimeZone` object.
+        """
+        default = self.app.config['ICU_DEFAULT_TIMEZONE']
+        if default is None:
+            default = 'UTC'
+        return (ICUtzinfo.getInstance(default).timezone)
+
+def load_messages(locale):
+    """Loads ICU messages for a given locale from the source files. Translation
+    files must be located in path: '<root>/translations/<lang>/'.
+    """
+    ctx = _request_ctx_stack.top
+    if ctx is None:
+        return None
+    dirname = os.path.join(ctx.app.root_path, TRANSLATIONS_PATH)
+    if not os.path.isdir(dirname):
+        raise Exception('Unable to find the translations path.')
+    locales_list = [name for name in os.listdir(dirname)
+                    if os.path.isdir(os.path.join(dirname, name))]
+    messages = {}
+    if locale not in locales_list:
+        raise Exception('No locale ICU message files found for the locale: %s' % locale)
+    else:
+        for subdir, dirs, files in os.walk(dirname + '/' + locale):
+            for file in files:
+                with open(subdir + '/' + file) as data_file:
+                    data = json.load(data_file)
+                    z = messages.copy()
+                    z.update(data)
+                    messages = z
+    return messages
+
+
+def get_message(key):
+    """Returns an ICU format message string for the given key."""
+    ctx = _request_ctx_stack.top
+    if ctx is None:
+        return None
+    messages = getattr(ctx, 'icu_messages', None)
+    if messages is None:
+        messages = get_messages()
+    if key in messages:
```

Comment:
_scoop before you loop_ (although hash lookups are cheap), it is more pythonic to replace this by:
```python
try:
    return messages[key]
except KeyError:
    return key
```